src/Testcontainers/Builders/ContainerBuilder`3.cs

```diff
@@ -329,6 +330,16 @@ public TBuilderEntity WithAutoRemove(bool autoRemove)
       return Clone(new ContainerConfiguration(autoRemove: autoRemove));
     }
 
+    /// <inheritdoc />
+    public TBuilderEntity WithReuse(bool reuse)
+    {
+      return Clone(new ContainerConfiguration(reuse: reuse)).WithCreateParameterModifier(createContainerParameters =>
+      {
+        createContainerParameters.Labels.Remove(TestcontainersClient.TestcontainersSessionIdLabel);
+        createContainerParameters.Labels.Remove(ResourceReaper.ResourceReaperSessionLabel);
```

Comment:
These labels are changing on each run. I'm not sure why they are in the `DefaultLabels` though. We need to remove them to keep the hash consistent.

They should be added via `WithCleanUp()` and `WithAutoRemove()`.

Reply:
> These labels are changing on each run. I'm not sure why they are in the `DefaultLabels` though. We need to remove them to keep the hash consistent.

Exactly. We need to ignore some properties while generating the hash value. Please see my more detailed answer below.

> They should be added via `WithCleanUp()` and `WithAutoRemove()`.

Yes, we should throw an exception for invalid configurations. I think we should do this in one of the `Validate()` base implementations.